Add docs about `@babel/core` support for `.cts` (#2722)
babel/website: docs/config-files.md

  file contains the [`"type": "module"`](https://nodejs.org/api/esm.html#esm_code_package_json_code_code_type_code_field)
  option, otherwise they are exactly the same as the `.cjs` files.

> - `babel.config.cts` and `.babelrc.cts` allow you to define your configuration as Typescript + CommonJS. They have been supported since `v7.21.0`. This is experimental support for Typescript integration, so requires you to have installed `@babel/preset-typescript` or running Babel using `ts-node`.
> 
> > It's not possible yet to use `babel.config.ts` and `babel.config.mts` files, pending stabilization of the Node.js ESM loader API.

Reviewer: ```suggestion
- `babel.config.cts` and `.babelrc.cts` allow you to define your configuration as Typescript + CommonJS. You must either install `@babel/preset-typescript`, or run Babel using `ts-node`.

> 🚧 This functionality is experimental. It's not possible yet to use `babel.config.ts` and `babel.config.mts` files, pending stabilization of the Node.js ESM loader API.
```
